src/pages/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useContext, useEffect, useState } from "react";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+
+import {
+  SearchForm,
+  GenreSelect,
+  MovieList,
+  MovieDetails,
+  SortControl,
+  AddEditMovieDialog,
+} from "../../components";
+import { MovieContext } from "@context/MovieContext";
+import { SortByOptions } from "@components/SortControl/SortControl.types";
+import { Button, ButtonTexts, Header } from "@shared/components";
+import styles from "./MovieListPage.module.scss";
+import { genresList } from "@shared/constants";
+import { IMovieInfo } from "@components/MovieList/MovieCard/MovieCard.types";
+
+const { addMovieButton, genreSortControls, mainContent, moviesNumber } = styles;
+
+const MovieListPage: React.FC = () => {
+  const [searchFormQuery, setSearchFormQuery] = useState<string>("");
+  const [activeGenre, setActiveGenre] = useState<string>(genresList[0].name);
+  const [sortCriterion, setSortCriterion] = useState<string>(
+    Object.keys(SortByOptions)[1],
+  );
+  const [movieList, setMovieList] = useState<IMovieInfo[]>([]);
+  const [isAddMovieModalOpened, setIsAddMovieModalOpened] =
+    useState<boolean>(false);
+  const { selectedMovie, setSelectedMovie } = useContext(MovieContext);
+
+  const fetchMovies = async (
```

Comment:
I believe would be better to move this fetch function to another folder. Try to keep UI in one layer, business logic in another, service in ...